### PR TITLE
fix missing host in location redirect

### PR DESCRIPTION
**Pull request by aheissenberger**
some servers return a location which is only the path. tall will fail without a hostname

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -22,7 +22,7 @@ export const tall = (url, options) => {
     return request({ method, protocol, host, path }, response => {
       if (response.headers.location && opt.maxRedirects) {
         opt.maxRedirects--
-        return resolve(tall(response.headers.location, opt))
+        return resolve(tall(response.headers.location.startsWith('http')?response.headers.location:`${protocol}//${host}${response.headers.location}`,opt))
```
Comment (lmammino) [inline src/index.js]: Being extra picky here, but what if the header contains something like: `http/website/index.html` ? Maybe we should check for `http://` and `https://` entirely.

**Comment by lmammino** [comment]
Hello (again 😁) @aheissenberger,
Thanks for this awesome PR. I left one comment in the review. Other than that I am super happy about this and I am looking forward to get this feature merged.

PS: Are you using `tall` in production? I'd be curious to know more about your use case.

File: src/index.js (hunk 1 above)
Comment (aheissenberger) [reply src/index.js]: There is allready code from you in the first part of tall to check for invalid URLs - no need to replicate.
The use case to cover is the url with an absolute path which is not allowed by definition but quit common.

**Comment by aheissenberger** [comment]
> Hello (again 😁) @aheissenberger,
> Thanks for this awesome PR. I left one comment in the review. Other than that I am super happy about this and I am looking forward to getting this feature merged.
> 
> PS: Are you using `tall` in production? I'd be curious to know more about your use case.

not in production - I had to export my "saved" links from google inbox and needed a quick solution to expand all the bookmarked urls before I sent them to Evernote with a simple AppleScript Code I found on the internet :-)

**Comment by lmammino** [comment]
Interesting use case. Glad you are finding this lib useful anyway. Getting this one merged shortly!

**Pull request merged**